utils/components.py

```diff
@@ -298,3 +301,302 @@ def ilab_importer_op(repository: str, release: str, base_model: dsl.Output[dsl.M
+                    # OCI Secret {sdg_oci_docker_secret} does not have an auth token present for {registry_name}. Ensure that the secret provided has the proper auth token #
+                    #########################################################################################################################################################
+                    """)
+                    sys.exit(1)
+                print(f"OCI Secret has auth token present for {registry_name}")
+            elif secret.type == "kubernetes.io/dockercfg":
+                # handle authentication if secret provided is kubernetes.io/dockercfg
+                dockercfg_json = json.loads(
+                    base64.b64decode(secret.data[".dockercfg"]).decode("utf-8")
+                )
+                if registry_name not in docker_config_json.keys():
+                    print(f"""
+                    ################################################################## ERROR ################################################################################
+                    # OCI Secret {sdg_oci_docker_secret} does not have an auth token present for {registry_name}. Ensure that the secret provided has the proper auth token #
+                    #########################################################################################################################################################
+                    """)
+                    sys.exit(1)
+                print(f"""
+                ######################## INFO ###########################
+                # OCI Secret has auth token present for {registry_name} #
+                #########################################################
+                """)
+        except ApiException as e:
+            print(f"""
+            ############################################## ERROR #################################################################
+            # Secret {sdg_oci_docker_secret} does not exist. Ensure you created a secret with this name in namespace {namespace} #
+            ######################################################################################################################
+            """)
+            sys.exit(1)
+
+
+@dsl.container_component
+def test_taxonomy_repo(sdg_repo_url: str):
+    return dsl.ContainerSpec(
+        RUNTIME_GENERIC_IMAGE,
+        ["/bin/sh", "-c"],
+        [
+            f"""
+            # Increase logging verbosity
+            set -x &&
+
+            # Add TLS Parameters if CA Cert exists and is non-zero size
+            ADDITIONAL_CLONE_PARAMS=""
+            if [ -s "$TAXONOMY_CA_CERT_PATH" ]; then
+                ADDITIONAL_CLONE_PARAMS="-c http.sslVerify=true -c http.sslCAInfo=$TAXONOMY_CA_CERT_PATH"
+            fi
+
+            # ls-remote will fail if repo is not valid
+            git ls-remote {sdg_repo_url} > /dev/null;
```

Comment:
Optional: It'd be nice to retry this on failure since you do it elsewhere in the code, but not required since it's easy enough to retry due to a network flake.